atc/db/team.go

```diff
@@ -788,6 +801,54 @@ func (t *team) FindCheckContainers(logger lager.Logger, pipelineName string, res
 	return containers, checkContainersExpiresAt, nil
 }
 
+type UpdateName struct {
+	OldName  string
+	NewName  string
+	TempName int
+}
+
+func (t *team) updateName(tx Tx, jobs []atc.JobConfig, pipelineID int) error {
+	jobsToUpdate := []UpdateName{}
+	counter := 0
+
+	for _, job := range jobs {
+		if job.OldName != "" {
+			jobsToUpdate = append(jobsToUpdate, UpdateName{
+				OldName:  job.OldName,
+				NewName:  job.Name,
+				TempName: counter,
+			})
+			counter += 1
+		}
+	}
+
+	for _, updateName := range jobsToUpdate {
+		_, err := tx.Exec(`
+	      UPDATE jobs
+	      SET name = $2
+	      WHERE name = $1 AND pipeline_id = $3
+	  `, updateName.OldName, updateName.TempName, pipelineID)
```

Comment:
What makes this `TempName` step necessary? Seems kind of confusing.

Reply:
I believe in the jobs table, the name column is unique. So in the case where a config has a name swap, there would be a conflict since it updates one row at a time... or is my understanding incorrect?

Reply:
Ah, gotcha, so it's for when you're doing something like this:

```yml
jobs:
- name: x
  old_name: y
- name: y
  old_name: x
```

Yeah I guess that's a bit tricky. I'm having kind of a hard time reasoning about a use case for that, though. To be honest I'd be OK with that just failing and requiring them to do it in two steps, since it makes the inner workings simpler. I'm a bit worried about those temporary numeric names. 🤷‍♂️

Reply:
That's a valid point. I couldn't really reason for that use case either but just thought of the worst possible scenario 😅. I'll remove this bit then